### PR TITLE
Add a timeout to the entrypoint wrapper

### PR DESCRIPTION
**Pull request by stevekuznetsov**
In order to give Prow a first-class concept of job timeout, we need to
add the option to timeout to the entrypoint wrapper.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/kind feature
/area prow
/cc @cjwagner @bentheelder
/assign @kryzacy
fixes https://github.com/kubernetes/test-infra/issues/7152

### CODE REVIEW

**Comment by BenTheElder** [comment]
It would also be great if sinker could determine when individual jobs were supposed to time out and terminate them, the job that triggered @krzyzacy's issue had a hung pod.

**Comment by krzyzacy** [comment]
this looks fine, +1 for sinker probably also should be aware of stale prowjobs

also you got a badge for actually typing my github handle and got it wrong :smirk:

**Comment by BenTheElder** [comment]
> wrong

right? :man_shrugging:

**Comment by stevekuznetsov** [comment]
Way too many Zs in that name @krzyzacy and auto-complete wasn't working ;)

@BenTheElder yeah we can add a timeout to sinker too, I guess, if we add the timeout to the ProwJob CRD itself in k8s? Why not have multiple layers of timeouts?

**Comment by stevekuznetsov** [comment]
@BenTheElder @krzyzacy any comments?

File: prow/cmd/entrypoint/main.go [hunk 1]
```diff
@@ -53,6 +55,7 @@ func gatherOptions() options {
 	o := options{}
 	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
 	o.wrapperOptions = wrapper.BindOptions(fs)
+	fs.DurationVar(&o.timeout, "timeout", 5*time.Hour, "Timeout for the test command.")
```
Comment (BenTheElder) [inline prow/cmd/entrypoint/main.go]: we won't be able to switch until this is properly tunable from prow config, scalability has 12+ hour jobs

Comment (stevekuznetsov) [reply prow/cmd/entrypoint/main.go]: I'd expect the first implementation of `prow` config that feeds `plank` to expose this. The first prototypes won't thread through config but will just explicitly create a complicated `PodSpec`, I think?

**Comment by stevekuznetsov** [comment]
Two holds makes one

/hold cancel

**Pull request merged**